airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceStrictEncrypt.java

```diff
@@ -39,6 +49,27 @@ public ConnectorSpecification modifySpec(final ConnectorSpecification originalSp
     return spec;
   }
 
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) throws Exception {
+    // #15808 Disallow connecting to db with disable, prefer or allow SSL mode when connecting directly
+    // and not over SSH tunnel
+    if (config.has(TUNNEL_METHOD)
+        && config.get(TUNNEL_METHOD).has(TUNNEL_METHOD)
+        && config.get(TUNNEL_METHOD).get(TUNNEL_METHOD).asText().equals(NO_TUNNEL)) {
+      // If no SSH tunnel
+      if (config.has(SSL_MODE) && config.get(SSL_MODE).has(MODE)) {
+        if (Set.of(SSL_MODE_DISABLE, SSL_MODE_ALLOW, SSL_MODE_PREFER).contains(config.get(SSL_MODE).get(MODE).asText())) {
+          // Fail in case SSL mode is disable, allow or prefer
+          return new AirbyteConnectionStatus()
+              .withStatus(AirbyteConnectionStatus.Status.FAILED)
```

Comment:
nit: in source-postgres-strict-encrypt it's slightly different
```          return new AirbyteConnectionStatus()
              .withStatus(Status.FAILED)
              .withMessage(
                  "Unsecured connection not allowed. If no SSH Tunnel set up, please use one of the following SSL modes: require, verify-ca, verify-full");
        }```

Reply:
> nit: in source-postgres-strict-encrypt it's slightly different
> 
> ```
>               .withStatus(Status.FAILED)
>               .withMessage(
>                   "Unsecured connection not allowed. If no SSH Tunnel set up, please use one of the following SSL modes: require, verify-ca, verify-full");
>         }```
> ```

now up-to-date with source-postgres-strict-encrypt